hawkbit-dmf/hawkbit-dmf-amqp/src/test/java/org/eclipse/hawkbit/integration/AmqpMessageDispatcherServiceIntegrationTest.java

```diff
@@ -603,4 +611,70 @@ private static boolean isDownloadAndUpdateRequest(final DmfMultiActionElement mu
                 || multiActionElement.getTopic().equals(EventTopic.DOWNLOAD_AND_INSTALL);
     }
 
+    @Test
+    @Description("Verify payload of batch assigments download and install message.")
+    public void assertBatchAssigmentsDownloadAndInstall() {
+        assertBatchAssigmentsMessagePayload(BATCH_DOWNLOAD_AND_INSTALL);
+    }
+
+    @Test
+    @Description("Verify payload of batch assigments download only message.")
+    public void assertBatchAssigmentsDownloadOnly() {
+        assertBatchAssigmentsMessagePayload(BATCH_DOWNLOAD);
+    }
+
+    protected void assertBatchAssigmentsMessagePayload(EventTopic topic) {
+        enableBatchAssignments();
+
+        List<String> targets = Arrays.asList("batchCtrlID1", "batchCtrlID2", "batchCtrlID3");
```

Comment:
Mark local variables as final to prevent re-assignments... more occurrences below

Reply:
done